mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -248,18 +248,18 @@ class ALSModel private[ml] (
   @Since("1.3.0")
   def setPredictionCol(value: String): this.type = set(predictionCol, value)
 
+  private val predict = udf { (userFeatures: Seq[Float], itemFeatures: Seq[Float]) =>
+    if (userFeatures != null && itemFeatures != null) {
+      blas.sdot(rank, userFeatures.toArray, 1, itemFeatures.toArray, 1)
```

Comment:
Good point! But since I copy-pasted this block in this PR, maybe it's okay to try it out in another PR? At least with what we have here we know it's not a regression. Want to make sure we get some version of ALS recommendForAll* in 2.2.